chips/sam4l/src/adc.rs

```diff
@@ -332,9 +354,21 @@ impl hil::adc::Adc for Adc {
                     return ReturnCode::FAIL;
                 }
             }
+
+            ReturnCode::SUCCESS
         }
+    }
+}
 
-        ReturnCode::SUCCESS
+/// Implements an ADC capable reading ADC samples on any channel.
+impl hil::adc::Adc for Adc {
+    type Channel = AdcChannel;
+
+    /// Enable and configure the ADC.
+    /// This can be called multiple times with no side effects.
+    fn initialize(&self) -> ReturnCode {
```

Comment:
(I also don't think the rest of the changes in this PR should block on this decision if there is disagreement)